services/src/main/java/org/keycloak/services/resources/admin/permissions/ModelRecord.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2025 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.services.resources.admin.permissions;
+
+import org.keycloak.authorization.AdminPermissionsSchema;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.GroupModel;
+import org.keycloak.models.RoleModel;
+import org.keycloak.models.UserModel;
+
+sealed interface ModelRecord {
```

Comment:
I would vote for having separate implementations of a `ResourceType` interface and better encapsulating the specific behaviors/state of each resource type.

Let's sync about this one, and depending on what we decide, we can do it as a follow-up.

Reply:
Will be done as follow-up